src/NextjsRevaluation.ts

```diff
@@ -0,0 +1,63 @@
+import { Duration, Stack } from 'aws-cdk-lib';
+import { Code, Function, FunctionOptions, Runtime } from 'aws-cdk-lib/aws-lambda';
+import { SqsEventSource } from 'aws-cdk-lib/aws-lambda-event-sources';
+import { Queue } from 'aws-cdk-lib/aws-sqs';
+import { Construct } from 'constructs';
+import { NextjsBaseProps } from './NextjsBase';
+import { NextjsBuild } from './NextjsBuild';
+import { NextJsLambda } from './NextjsLambda';
+
+export interface RevaluationProps extends NextjsBaseProps {
+  /**
+   * Override function properties.
+   */
+  readonly lambdaOptions?: FunctionOptions;
+
+  /**
+   * The `NextjsBuild` instance representing the built Nextjs application.
+   */
+  readonly nextBuild: NextjsBuild;
+
+  /**
+   * The main NextJS server handler lambda function.
+   */
+  readonly serverFunction: NextJsLambda;
+}
+
+/**
+ * Builds the system for revaluating Next.js resources. This includes a Lambda function handler and queue system.
+ */
+export class NextjsRevaluation extends Construct {
+  constructor(scope: Construct, id: string, props: RevaluationProps) {
+    super(scope, id);
+
+    if (!props.nextBuild) return;
+
+    const code = props.isPlaceholder
+      ? Code.fromInline(
+          "module.exports.handler = async () => { return { statusCode: 200, body: 'SST placeholder site' } }"
+        )
+      : Code.fromAsset(props.nextBuild.nextImageFnDir);
+
+    const queue = new Queue(this, 'RevalidationQueue', {
+      fifo: true,
+      receiveMessageWaitTime: Duration.seconds(20),
+    });
+    const consumer = new Function(this, 'RevalidationFunction', {
+      description: 'Next.js revalidator',
+      handler: 'index.handler',
+      code,
+      runtime: Runtime.NODEJS_18_X,
+      timeout: Duration.seconds(30),
+      // This, I think, is supposed to be the VPC or VPC Subnet config (https://github.com/serverless-stack/sst/blob/master/packages/sst/src/constructs/NextjsSite.ts#L59C5-L59C17)
```

Comment:
I don't see why this function would need a VPC. Although, we maybe should add props to enable a developer to add this function into a VPC if they want to but I'm not sure why they would.

Reply:
I'm not actually sure either to be honest. Honestly I don't know, can an SQS queue be "in" a VPC? I'd guess that's not relevant. I'll remove this for now.

Reply:
@kevin-mitchell, SQS doesn't run in a VPC. It's a managed service so you don't have control over where it runs.

Reply:
👍 , that was my understanding